Add Flow Diagrams to Guides (#277)
figment-networks/figment-documentation: docs/guides/staking-api/Solana/merge-stake/index.mdx

pagination_prev: null
---

import Link from "@docusaurus/Link";
import img1 from "@site/static/img/staking/workflows/v1/solana/Workflows::V1::Solana::MergeStakeAccountFlow_workflow.png";

## Introduction

The Merge Stake Account flow allows you to merge two stake accounts that have the same authorities and lockup, into a single stake account.
A merge is possible between two stakes in the following states with no additional conditions:

- Two deactivated stakes
- An inactive stake into an activating stake during its activation epoch

For the following cases, the voter public key and vote credits observed must match:

- Two activated stakes
- Two activating accounts that share an activation epoch, during the activation epoch

All other combinations of stake account states will fail to merge, including all "transient" states
(where a stake is activating or deactivating with a non-zero effective stake).

To merge two separate stake accounts into one, you will need:

- `destination_stake_account_pubkey` - Public key of the stake account into which you want to merge another stake account

> - `source_stake_account_pubkey` - Public key of the stake account which you want to merge with the another stake account

Reviewer: ```suggestion
- `source_stake_account_pubkey` - Public key of the stake account which you want to merge into a destination stake account
```
